server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java

```diff
@@ -187,4 +189,25 @@ IndexMetadata archiveBrokenIndexSettings(IndexMetadata indexMetadata) {
             return indexMetadata;
         }
     }
+
+    /**
+     * Convert shared_cache searchable snapshot indices to only specify
+     * _tier_preference: data_frozen, removing any pre-existing tier allocation rules.
+     */
+    IndexMetadata convertSharedCacheTierPreference(IndexMetadata indexMetadata) {
+        final Settings settings = indexMetadata.getSettings();
+        // Only remove these settings for a shared_cache searchable snapshot
+        if ("snapshot".equals(settings.get("index.store.type", "")) && settings.getAsBoolean("index.store.snapshot.partial", false)) {
+            final Settings.Builder newSettings = Settings.builder().put(settings);
+            // Clear any allocation rules other than preference for tier
+            newSettings.remove("index.routing.allocation.include._tier");
+            newSettings.remove("index.routing.allocation.exclude._tier");
+            newSettings.remove("index.routing.allocation.require._tier");
+            // Override the tier preference to be only on frozen nodes, regardless of its current setting
+            newSettings.put("index.routing.allocation.include._tier_preference", "data_frozen");
+            return IndexMetadata.builder(indexMetadata).settings(newSettings.build()).build();
```

Comment:
I think we would want to check if the settings are already OK and then return the original instance in order to not rebuild the Metadata completely every time?

Reply:
Sure, it only does it once, but it makes sense not to do it at all if needed. I changed this and added a test